### PR TITLE
User auth api

### PR DESCRIPTION
**Pull request by wgreenberg**

### CODE REVIEW

File: backend/src/api/board.rs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+use actix_web::{ post, web, HttpResponse, Responder };
```
Comment (klautcomputing) [inline backend/src/api/board.rs:1]: nitpick, `cargo fmt` as your indentation is inconsistent :)

File: backend/src/api/user.rs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+use actix_web::{ get, post, web, Responder };
+use serde::Deserialize;
+
+use crate::extractors::auth::AuthenticatedUser;
+use crate::model::user::User;
+use crate::db::util::DbPool;
+
+#[get("/user/{uid}")]
+pub async fn get_user(uid: web::Path<String>, pool: web::Data<DbPool>) -> impl Responder {
+    User::find_by_uid(pool.get_ref(), uid.as_ref()).await
```
Comment (klautcomputing) [inline backend/src/api/user.rs:11]: For later: What do you think about making a function for this? Since, we will use this in a lot of places I imagine.

**Review by klautcomputing — APPROVED** [review]
Nice thank you! Just a small things, feel free to merge this at any point.

File: backend/src/model/user.rs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    QueryDsl,
+    insert_into,
+};
+use serde::{Serialize, Deserialize};
+use crate::db::util::{get_conn, DbPool};
+use crate::db::schema::users;
+use crate::db::schema::users::dsl::users as users_table;
+use diesel_async::RunQueryDsl;
+
+#[derive(Queryable, Insertable, Serialize, Deserialize)]
```
[the remark this answers is not in the document]
Comment (wgreenberg) [reply backend/src/model/user.rs:14]: oh nice, i hadn't noticed that trait

File: backend/src/api/board.rs (hunk 1 above)
Comment (wgreenberg) [reply backend/src/api/board.rs]: so it is! added a commit w/ `cargo fmt` changes

File: backend/src/api/user.rs (hunk 2 above)
Comment (wgreenberg) [reply backend/src/api/user.rs]: i suspect you're right, but i tend to wait until refactoring like that's needed before writing it -- DRY doesn't count if we haven't repeated ourselves yet :P

**Review by klautcomputing — APPROVED** [review]
I should make the rules a little less strict :+1:

**Pull request merged**